spec/docs/joinconditions.md

```diff
@@ -96,3 +96,77 @@ then the `rml:child` shortcut could be used.
     rml:logicalSource <LS2> ;
     rml:subjectMap <#SM2> .
 ```
+## Join types
+
+If the [Logical Source]() of the [Triples Map]() that contains the [Referencing Object Map]() 
+and the [Logical Source]() of the [Referencing Object Map]()'s [Parent Triples Map]() are not identical, 
+then the referencing object map must have at least one join condition.
+
+A [Logical Source]() is considered as identical to another [Logical Source]() 
+when the set of objects at the end of the property paths starting with `rml:source` and starting with `rml:iterator` are identical. 
```

Comment:
Why property paths? 

```suggestion
when
* the value of the source property (`rml:source`) of both logical sources is equal, and
* the value of the reference formulation property (`rml:referenceFormulation`) of both logical sources is equal, and
* the value of the iterator property (`rml:iterator`) of both logical sources is equal, or both logical sources do not specify the iterator property.
```

Reply:
@pmaria I couldn't think of another way to specify that also the descriptions of nested sources should be equals (so nested source descriptions can have different identifiers, but still have the same values for all nested properties (e.g. <LS1> and <LS2> are equal in the example, even is they have different identifiers.)

Reply:
as discussed today:

```
when
* the value of the reference formulation property (`rml:referenceFormulation`) of both logical sources is equal,
* the value of the iterator property (`rml:iterator`) of both logical sources is equal, or both logical sources do not specify the iterator property, and
* the sub RDF graphs of the source property (`rml:source`) of both logical sources that only contain RML actionable properties of the source access descriptions are isomorph.
```

this last point means that each source access description needs to list its actionable properties